src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -67,15 +67,26 @@ def generate_mobile_frame(parameters: Dict[str, Optional[str]]) -> Dict[str, str
     return frame
 
 
-def set_top_tags(scope: Scope, project: Project) -> None:
-    scope.set_tag("project.slug", project.slug)
-    scope.set_tag("organization.slug", project.organization.slug)
+def set_top_tags(
+    scope: Scope,
+    project: Project,
+    ctx: Mapping[str, Optional[str]],
+    has_code_mappings: bool,
+) -> None:
     try:
-        ea_org: bool = project.organization.flags.early_adopter.is_set
-        scope.set_tag("organization.early_adopter", ea_org)
+        scope.set_tag("project.slug", project.slug)
+        scope.set_tag("organization.slug", project.organization.slug)
+        scope.set_tag(
+            "organization.early_adopter", bool(project.organization.flags.early_adopter.is_set)
+        )
+        scope.set_tag("stacktrace_link.platform", ctx["platform"])
+        scope.set_tag("stacktrace_link.has_code_mappings", has_code_mappings)
+        if ctx["platform"] == "python":
+            # This allows detecting a file that belongs to Python's 3rd party modules
+            scope.set_tag("stacktrace_link.in_app", "site-packages" in ctx["file"])
```

Comment:
When a 3rd party modules is considered as an in-app frame, we will hit the BE and we will respond saying that we have an invalid code mapping which we don't. This will help us validate my theory.